src/events/tls.py

```diff
@@ -164,3 +172,11 @@ def _set_tls_private_key(self, event: ActionEvent) -> None:
 
         self.charm.state.unit_server.update({"private-key": private_key})
         self._on_certificate_expiring(event)
+
+    def _cleanup_old_ca_field(self) -> None:
+        """In order to ensure backwards compatibility, we keep old secrets until the first time they are updated.
+
+        This will allow to safely roll back soon after an upgrade.
+        """
+        if self.charm.state.unit_server.relation_data.get("ca"):
+            self.charm.state.unit_server.update({"ca": ""})
```

Comment:
**todo:** I think this can be removed.

Reply:
My logic here was this: deleting anything with DP libs is  a "big machinery".

Elegnt call like:
```
self.charm.state.unit_broker.update( {"certificate": event.certificate, "ca-cert": event.ca, "ca": ""} )
```
would result in quite a sequence of executions just to discover that `ca` never even existed.

It's possible to go this way (as long as we see no performance penalty -- we shouldn't, not suppsoed to be "that" bad).
Anyhow, this was the rationale behind

Reply:
Yeah I can see that, I guess it's the same warning as 'Trying to remove non-existent key' right?

I actually quite like the ability to just endlessly 'wipe' keys even if they're not there, so if that's possible we should go that way.

Reply:
Sure, we can. If at any point (i.e. libs changing, etc.) you may observe any performance issues, it may worth to revisit this code.

Reply:
(The above comment implicitly refers to time delays we've initially experienced with pulling secrets, typically. While the current implementation should make these 2 solutions close enough in terms of execution time, that relies on the fact that currently we're caching (snapshotting) secrets for the time of the event handler. Typically once that goes away -- which has a chance to happen soon-- we may again fully rely on `ops` workflows on secret access.)

Reply:
Done.

Reply:
Note that each time the call is invoked, you will get this in the logs:

```
WARNING  charms.data_platform_libs.v0.data_interfaces:data_interfaces.py:1152 Non-existing field 'ca' was attempted to be removed from the databag (relation ID: 0)
```